File: scope104/src/main/java/com/petro/scope104/util/SquareCardView.java
package com.petro.scope104.util;

import android.content.Context;
import android.util.AttributeSet;

import androidx.cardview.widget.CardView;

public class SquareCardView extends CardView {

    public SquareCardView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int minSize = Math.min(widthMeasureSpec, heightMeasureSpec);
        super.onMeasure(minSize, minSize);
    }

}

File: scope104/src/main/kotlin/com/petro/scope104/util/SquareCardView.kt
package com.petro.scope104.util

import android.content.Context
import android.util.AttributeSet
import androidx.cardview.widget.CardView

class SquareCardView(context: Context?, attrs: AttributeSet?) : CardView(
    context!!, attrs
) {
    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val minSize = Math.min(widthMeasureSpec, heightMeasureSpec)
        super.onMeasure(minSize, minSize)
    }
}